### PR TITLE
Use new Rust based ad-block library

### PR DESCRIPTION
**Pull request by bbondy**
Fix https://github.com/brave/brave-browser/issues/4793
Please let me merge if approved, I still need to verify CI.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [x] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_shields/browser/ad_block_base_service.cc [hunk 1]
```diff
@@ -33,70 +33,66 @@ using namespace net::registry_controlled_domains;  // NOLINT
...
-      filter_option = FOXmlHttpRequest;
+      filter_option = "xhr";
       break;
     // a ping request for <a ping>/sendBeacon.
     case content::ResourceType::kPing:
-      filter_option = FOPing;
+      filter_option = "ping";
       break;
-    // the main resource of a dedicated worker.
+    // the main resource of a dedicated
```
Comment (emerick) [inline components/brave_shields/browser/ad_block_base_service.cc]: Nitpick: Seems like the end of this comment went missing...

File: components/brave_shields/browser/ad_block_base_service.cc [hunk 2]
```diff
@@ -33,70 +33,66 @@ using namespace net::registry_controlled_domains;  // NOLINT
...
     case content::ResourceType::kWorker:
     // the main resource of a shared worker.
     case content::ResourceType::kSharedWorker:
     // an explicitly requested prefetch
     case content::ResourceType::kPrefetch:
     // the main resource of a service worker.
     case content::ResourceType::kServiceWorker:
-    // a report of Content Security Policy violations.
+    // a report of Content Security Policy
```
Comment (emerick) [inline components/brave_shields/browser/ad_block_base_service.cc]: This comment is missing the end too? Or maybe it's intentional, I can't tell. :-)

**Comment by bbondy** [comment]
> (and I can't build locally on mac at the moment)

Do you mean you can't build this PR in particular? Or you can't build at all?

> if the signing script needs to be patched to have it signed.

There's a commit to do the signing of it already. I haven't verified on CI if that works yet though.

**Comment by mkarolin** [comment]
> > (and I can't build locally on mac at the moment)
> 
> Do you mean you can't build this PR in particular? Or you can't build at all?
> 
Just meant I was on Windows and couldn't switch to Mac at that moment.

**Comment by simonhong** [comment]
> There's a commit to do the signing of it already. I haven't verified on CI if that works yet though.

@bbondy I think you missed one thing in that commit.
See https://github.com/brave/brave-core/blob/master/patches/chrome-installer-mac-signing-signing.py.patch#L26

**Pull request merged**